### PR TITLE
DB-9845 Spark 3.0 Migrate org.apache.spark.Accumulable to AccumulatorV2

### PR DESCRIPTION
**Pull request by ipraznik-splice**

### CODE REVIEW

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1059/

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1063/

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

File: hbase_sql/src/main/java/com/splicemachine/stream/accumulator/BadRecordsAccumulator.java [hunk 1]
```diff
@@ -14,34 +14,54 @@
...
+
+    @Override
+    public boolean isZero() {
+        return badRecordsRecorder.getNumberOfBadRecords() == 0;
+    }
+
+    @Override
+    public AccumulatorV2<String, BadRecordsRecorder> copy() {
+        return new BadRecordsAccumulator(badRecordsRecorder);
+    }
```
Comment (martinrupp) [inline hbase_sql/src/main/java/com/splicemachine/stream/accumulator/BadRecordsAccumulator.java:46]: if i understand correctly, you might as well return **this** here:
BadRecordsAccumulator only has one internal state, namely the badRecordsRecorder. With this code, every "copy" points to the same badRecordsRecorder, so will have a shared state, making copies not really copies. e.g. everything you would add or merge to one BadRecordsAccumulator, you would also see in the copy. Not sure if this is intended.

Comment (ipraznik-splice) [reply hbase_sql/src/main/java/com/splicemachine/stream/accumulator/BadRecordsAccumulator.java]: you are right, thanks. I've improved the copy method, added reset for BadRecordsRecorder as well

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1109/

**Comment by ipraznik-splice** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST SUCCEEDED +1

**Comment by martinrupp** [comment]
jenkins please test branch @cdh6.3.0,skipTestsLongerThan2Minutes

**Comment by spliceci** [comment]
TEST FAILED
http://nexus.splicemachine.com:8080/job/spliceengine-PR/job/Splice-PR-Branch/1233/

**Comment by martinrupp** [comment]
jenkins +1 sporadic testMultipleMerge

**Pull request merged**